File: complete_search/full_solution/some_sum.py
n = int(input())

even = False
odd = False
for i in range(1, 100 + 1):
    if i + n - 1 <= 100:
        sum = (n / 2) * (i + i + n - 1) # could have used a for loop instead
        if sum % 2 == 0:
            even = True
        else:
            odd = True

print("Either" if even and odd else "Even" if even else "Odd")

File: complete_search/full_solution/some_sum.cpp
#include <iostream>

using namespace std;

int main() {
    int n, sum;
    bool even = false, odd = false;
    cin >> n;
    for (int i = 1; i <= 100; i++) {
        if (i + n - 1 <= 100) {
            sum = (n / 2.0) * (i + i + n - 1); // could have used loop instead
            if (sum % 2 == 0)
                even = true;
            else
                odd = true;
        }
    }
        
    if (even && odd)
        cout << "Either";
    else if (even)
        cout << "Even";
    else
        cout << "Odd";

    return 0;
}